### ISSUE TITLE
Investigate feasibility of Windows support

### ISSUE THREAD

**Issue by ezramorris** [ezramorris/PyVirtualSerialPorts#3]
Investigate methods that could potentially be used to add Windows support, for instance:

- [Windows Pseudoconsoles](https://learn.microsoft.com/en-us/windows/console/creating-a-pseudoconsole-session)
- A driver like [com0com](https://com0com.sourceforge.net/)

**Comment by kevinlinxc** [comment]
Agreed, this package could be a really good abstract interface if it worked on all platforms

**Comment by ezramorris** [comment]
I've had a look in to a few options.

Whilst Windows Psuedoconsole looks promising, the implementation is that when you create a psuedoconsole, you get a handle which you then pass as part of application startup, therefore it's not a suitable implementation for programs expecting to talk to serial ports.

com0com is probably the most appropriate open source option, however there are some notable drawbacks, namely that it only has a CLI interface (no API) and requires running as administrator. Not insurmountable, however.

**Comment by ezramorris** [comment]
Decision:

- use com0com for Windows support
- implement Python wrapper of com0com, as it may be useful on its own too
